File: string/58.go
func lengthOfLastWord(s string) int {
    length := len(s)
    var flag bool
    var lengthEndSpace int
    for i := length - 1; i >= 0; i-- {
        if s[i] == ' ' {
            if flag {
                return length - i - 1 - lengthEndSpace   
            }else {
                lengthEndSpace++
            }
        }else {
            flag = true
        }
    }
    return length - lengthEndSpace
}

File: string/58.cc
class Solution {
public:
    int lengthOfLastWord(string s) {
        int ret = 0;
        bool flag = false;
        for (int i = s.size()-1; i >= 0; i--) {
            if (s[i] != ' ') {
                flag = true;
                ret++;
            } else {
                if (!flag) {
                    continue;
                } else {
                    break;   
                }
            }
        }
        return ret;
    }
};
